### ISSUE TITLE
Packagekit Updates: not clear what to do if the system is not subscribed

### ISSUE THREAD

**Issue by andreasn** [cockpit-project/cockpit#7194]
If I'm on a RHEL system and I haven't yet subscribed the system, trying to update the system via the updates page won't work.
We should warn about this somehow, guiding the administrator that they need to subscribe the system first.

**Comment by andreasn** [comment]
What OSTree updates does:
![screenshot from 2017-07-11 15-55-54](https://user-images.githubusercontent.com/209895/28072625-e55e6df6-6653-11e7-890e-3f7d275f8d6a.png)

What the CLI-tool does:
![screenshot from 2017-07-07 21-25-10](https://user-images.githubusercontent.com/209895/28072663-faabe7ce-6653-11e7-95da-9f3ed3183308.png)

**Comment by garrett** [comment]
There are two states where this might happen:

1. new machines, but unsubscribed RHEL
2. previously used machine (with update info/pending/history), but lapsed RHEL subscription

We will focus on state 1 first.

**Comment by martinpitt** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Software Updates: upgrade log, layout, RHEL registration](https://trello.com/c/w0weB970/534-software-updates-upgrade-log-layout-rhel-registration)

**Comment by garrett** [comment]
## When there are no pending updates (fresh system or registration has lapsed and there are no updates)

![unregistered-empty-state](https://raw.githubusercontent.com/cockpit-project/cockpit-design/master/software-updates/software-updates-packages-unregistered-empty.png)

There are three choices:

1. **Full integration** of the registration process in the software tab (importing the registration tab).
2. **There-and-back-again**: A guided adventure to the registration tab, with a path back.
3. **Switch**: Only a way to get to the registration page. Once you're there, you're on your own.

Concept 1 would be best, but would require a bunch of work and might not be possible in Cockpit without some heavy lifting.

Concept 3 is not useful. It's better than what we have right now, but not by much.

Concept 2 is the middle ground. It would require a little bit of work in the registration page, but it should be called in such a way that clicking a button on the software page:

1. jumps to the registration page
2. automatically triggers the registration dialog (unless there's something I'm not aware of in the registration tab)
3. has a link back to the software page (basically a "next-step")

## When the system's registration has lapsed and there are pending updates

![lapsed registration](https://raw.githubusercontent.com/cockpit-project/cockpit-design/master/software-updates/software-updates-packages-registration_lapse.png)

As there could be multiple reasons why things might not work, this button would simply switch to the page to allow someone to inspect the current status and fix it.

Ideally, we'd use a bit of concept 2 (above, aka: "there-and-back-again") to have a link back to software.

**Comment by martinpitt** [comment]
@andreasn: How did you get this screen? Our test images are unregistered, and indeed I can't get any software updates on our rhel-7 images. But I can get OSTree updates on rhel-atomic, and I don't see the "This system is not registered ..." error on the OSTree page. Nor do any of the strings `unconfigured-state`, `not registered`, or `Subscription Management` appear in Cockpit. Maybe this is something that gets patched in downstream?

**Comment by martinpitt** [comment]
I. e. I'd like to re-use the code (not how it looks like) for checking whether a system is subscribed.

**Comment by martinpitt** [comment]
@andreasn: I checked the rhel-7.3, -7.4, and atomic branches in the rhel dist-git, and there are no patches to that effect, so I'd really like to see where that came from. Thanks!

**Comment by martinpitt** [comment]
I'm going to use the D-Bus call that Cockpit's 
[subscription-manager](https://github.com/cockpit-project/cockpit/blob/master/pkg/subscriptions/subscriptions-client.js#L282) uses, which is also the one documented in [candlepin](http://www.candlepinproject.org/docs/subscription-manager/dbus.html). This is simple and straightforward.

**Comment by martinpitt** [comment]
@garrett: I implemented the simplest option for now, which is already better than the status quo. I'll see to getting a link from subscriptions back to updates in a separate PR. Tests are still missing (thus no PR yet), but I have two questions/comments.

The "fresh system" screen shot is very close to your design:

![unregistered-fresh](https://user-images.githubusercontent.com/200109/28822055-1bbea6a8-76b8-11e7-9f85-92d6bacef3ca.png)

I. e. same text, almost the same layout, and hidden header bar. The only difference is that the "Red Hat Customer Portal" is not a link. I was wondering if we need to encourage/hint people to go there instead of the Cockpit Subscriptions page. If we do want the link, should that go to https://access.redhat.com/ or somewhere more specific?

The "expired" case looks like this:

![unregistered-available](https://user-images.githubusercontent.com/200109/28822104-50a3fb52-76b8-11e7-9e56-ae2244e222c9.png)

I. e. warning in a separate heading, with a primary button (that again goes to Subscriptions), hidden actions (Check for updates, Apply) and the available updates are still being shown. I just changed the text to be two sentences, as I don't see how to split a sentence in the middle (one `<strong>` part and the other is not) without giving translators a really hard time. Suggestions for changing the verbiage is appreciated, of course!

**Comment by martinpitt** [comment]
I filed issue #7454 about improving the interaction between updates and subscriptions.

**Comment by garrett** [comment]
I'm fine with it not being a link. 

The reason I made it a link in the mockup is because I thought it might be nice to be able to point people to the right place in case there's any problem. 

It would handle the cases where a person:
- would need to reactivate lapsed subscriptions (where this particular machine is not subscribed yet) 
- needs quick pointer to the place to subscribe if this happens to be the first RHEL machine they've used
- needs reference to their credentials for the subscription dialog

It's probably better in the subscription page overall, unless the flow would eventually somehow show a dialog in this page. (Basically: The topmost set of mockups out of the three.) 

In the current (and probable future) incarnation, the link probably doesn't make as much sense in the software page.

_**TL;DR**: Leaving out the link is fine. 👍_
